integration_tests/tests/in_cluster.rs

```diff
@@ -0,0 +1,420 @@
+        // service port numbers. The leader uses its view of its own endpoint URL to construct
+        // collection job URIs, so we will only patch each aggregator's view of its peer's endpoint.
+        let leader_endpoints = {
+            let mut endpoints = leader_task.aggregator_endpoints().to_vec();
+            endpoints[1] = Self::in_cluster_aggregator_url(&helper_namespace);
+            endpoints
+        };
+        let leader_task = leader_task
+            .with_aggregator_endpoints(leader_endpoints)
+            .build();
+        let leader = InClusterJanus::new(
+            &cluster,
+            &leader_namespace,
+            &leader_api_namespace,
+            &leader_task,
+        )
+        .await;
+
+        let helper_endpoints = {
+            let mut endpoints = helper_task.aggregator_endpoints().to_vec();
+            endpoints[0] = Self::in_cluster_aggregator_url(&leader_namespace);
+            endpoints
+        };
+        let helper_task = helper_task
+            .with_aggregator_endpoints(helper_endpoints)
+            .build();
+        let helper = InClusterJanus::new(
+            &cluster,
+            &helper_namespace,
+            &helper_api_namespace,
+            &helper_task,
+        )
+        .await;
+
+        Self {
+            leader_task,
+            collector_private_key,
+            leader,
+            helper,
+        }
+    }
+
+    fn in_cluster_aggregator_url(namespace: &str) -> Url {
+        Url::parse(&format!(
+            "http://aggregator.{namespace}.svc.cluster.local:80"
+        ))
+        .unwrap()
+    }
+}
+
+struct InClusterJanus {
+    aggregator_port_forward: PortForward,
+}
+
+impl InClusterJanus {
+    /// Set up a port forward to an existing Janus instance in a Kubernetes cluster, and provision a
+    /// DAP task in it via divviup-api.
+    async fn new(
+        cluster: &Cluster,
+        aggregator_namespace: &str,
+        control_plane_namespace: &str,
+        task: &Task,
+    ) -> Self {
+        let divviup_api_port_forward = cluster
+            .forward_port(control_plane_namespace, "divviup-api", 80)
+            .await;
+        divviup_api_create_task(&divviup_api_port_forward, task).await;
+        let aggregator_port_forward = cluster
+            .forward_port(aggregator_namespace, "aggregator", 80)
+            .await;
+        Self {
+            aggregator_port_forward,
+        }
+    }
+
+    fn port(&self) -> u16 {
+        self.aggregator_port_forward.local_port()
+    }
+}
+
+// Serialization/deserialization helper structures for interaction with divviup-api.
+
+#[derive(Deserialize)]
+struct Account {
+    id: String,
+}
+
+#[derive(Serialize)]
+struct Histogram {
+    buckets: Vec<u64>,
+}
+
+#[derive(Serialize)]
+struct Sum {
+    bits: usize,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "snake_case", tag = "type")]
+enum ApiVdaf {
```

Comment:
Seems odd to me that this integration test package should have to define API request types for `divviup-api`. Is there a plan to export these somehow from `divviup-api`? Or would that create a dependency cycle?

Reply:
I hadn't considered that, but it would have to be via a git dependency, unless we want to publish divviup-api (or part of it) to crates.io. Since we can only release all Janus crates at the same time, if there were to be future breaking changes to `HpkeConfig`, we might have to disable the integration test for a release, but otherwise dependency cycles wouldn't be a concern.

Reply:
Let's move forward with this change as-is, and we can consider publishing a Rust client for `divviup-api` separately.